File: frontend/src/app/model/natuurlijk-persoon-fiche-dto.ts
import {NatuurlijkPersoonDto} from "./natuurlijk-persoon-dto";
import {RelatieDto} from "./relatie-dto";
import {BroerOfZusDto} from "./broer-of-zus-dto";
import {NonkelOfTanteDto} from "./nonkel-of-tante-dto";

export class NatuurlijkPersoonFicheDto {
  public constructor(public natuurlijkPersoon: NatuurlijkPersoonDto,
                     public moeder: NatuurlijkPersoonDto | null,
                     public vader: NatuurlijkPersoonDto | null,
                     public relaties: RelatieDto[],
                     public kinderen: NatuurlijkPersoonDto[],
                     public broersEnZussen: BroerOfZusDto[],
                     public nonkelsEnTantes: NonkelOfTanteDto[],
                     public nevenEnNichten: NatuurlijkPersoonDto[],
                     public documenten: any[]) {
  }
}

File: domein/src/main/java/be/genealogie/domein/dto/NatuurlijkPersoonFicheDto.java
package be.genealogie.domein.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class NatuurlijkPersoonFicheDto {
    private NatuurlijkPersoonDTO moeder;
    private NatuurlijkPersoonDTO vader;
    private NatuurlijkPersoonDTO natuurlijkPersoon;
    private List<RelatieDto> relaties;
    private List<NatuurlijkPersoonDTO> kinderen;
    private List<NatuurlijkPersoonDTO> broersEnZussen;
    private List<NatuurlijkPersoonDTO> nevemEnNichten;
    private List<NatuurlijkPersoonDTO> oomsEnTantes;
    private List<DocumentHeaderDto> documenten;
}
